services/director-v2/src/simcore_service_director_v2/utils/dags.py

```diff
@@ -224,34 +225,29 @@ def compute_pipeline_submitted_timestamp(
     node_id_to_comp_task: dict[NodeIDStr, CompTaskAtDB] = {
         NodeIDStr(f"{task.node_id}"): task for task in comp_tasks
     }
-    pipeline_submitted_at = max(
-        node_id_to_comp_task[node_id].submit for node_id in pipeline_dag.nodes
-    )
-
-    return pipeline_submitted_at
+    return max(node_id_to_comp_task[node_id].submit for node_id in pipeline_dag.nodes)
 
 
 async def compute_pipeline_details(
     complete_dag: nx.DiGraph, pipeline_dag: nx.DiGraph, comp_tasks: list[CompTaskAtDB]
 ) -> PipelineDetails:
-    try:
+    with contextlib.suppress(nx.exception.NetworkXUnfeasible):
         # NOTE: this problem of cyclic graphs for control loops create all kinds of issues that must be fixed
         # first pass, traversing in topological order to correctly get the dependencies, set the nodes states
         await _set_computational_nodes_states(complete_dag)
-    except nx.exception.NetworkXUnfeasible:
-        # not acyclic
-        pass
 
     # NOTE: the latest progress is available in comp_tasks only
     node_id_to_comp_task: dict[NodeIDStr, CompTaskAtDB] = {
         NodeIDStr(f"{task.node_id}"): task for task in comp_tasks
     }
     pipeline_progress = None
     if len(pipeline_dag.nodes) > 0:
-        pipeline_progress = 0.0
-        for node_id in pipeline_dag.nodes:
-            if node_progress := node_id_to_comp_task[node_id].progress:
-                pipeline_progress += node_progress / len(pipeline_dag.nodes)
+        pipeline_progress = sum(
+            node_id_to_comp_task[node_id].progress / len(pipeline_dag.nodes)
```

Comment:
MINOR: you can save computing `len(pipeline_dag.nodes)` everytime in the loop if you store it when you check the `if`,

ie. 

```python

if nodes_count : = len(pipeline_dag.nodes)
      pipeline_progresss = sum( node_id_to_comp_task[node_id].progress / nodes_count for node_id in pipeline_dag.nodes if node_id_to_comp_task[node_id].progress is not None )
   # ...
```